pio/seven_segment/pio_7_segment_library/7_segment_lib.c

```diff
@@ -0,0 +1,75 @@
+#include <stdio.h>
+#include "hardware/gpio.h"
+#include "7_segment_lib.h"
+
+// segment patterns for digits 0-9
+static const uint8_t segments[] = {
+//    EDBGACF.      NB: these bit patterns should match the way you connect the GPIOs
+    0b11101110,     // 0
+    0b00100100,     // 1
+    0b11111000,     // 2
+    0b01111100,     // 3
+    0b00110110,     // 4
+    0b01011110,     // 5
+    0b11011110,     // 6
+    0b00101100,     // 7
+    0b11111110,     // 8
+    0b01111110      // 9
+};
+
+
+// converts an integer into a 32-bit word representing up to four 7-segment digits
+//
+uint32_t int_to_seven_segment (int num) {
+    uint32_t word = 0;
+    if (num == 0) {
+        word = segments[0];
+    } else {
+        for (int bitshift = 0; bitshift < 32 && num > 0; bitshift += 8) {
+            word |= segments[num % 10] << bitshift;
```

Comment:
I think this might be slightly more readable if you rearranged it to:
```suggestion
        for (int digit = 0; digit < 4 && num > 0; digit++) {
            word |= segments[num % 10] << (digit * 8);
```
?

Reply:
Hmmm - sorry to disagree but I think mine's more readable (and I know it works)... personal taste I guess, unless you feel strongly?

Reply:
> personal taste I guess

Fair enough :+1: